cluster-assign-roles.sh

```diff
@@ -181,10 +181,24 @@ function hadoop_install {
   done
 
   printf "Installing workers...\n"
+  status_file=${TMPDIR:-/tmp}/status_file_$$
+  function clean_up_status_file {
+    rm -f $status_file
+  }
+  trap clean_up_status_file EXIT
+  tempfile -n $status_file
```

Comment:
`tempfile` command is not recognized by the following bash versions:
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin14)
GNU bash, version 3.2.25(1)-release (x86_64-redhat-linux-gnu)

On, both versions, `mktemp $status_file` works.